lib/rules/no-restricted-service-injections.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+const kebabCase = require('lodash.kebabcase');
+const assert = require('assert');
+const emberUtils = require('../utils/ember');
+
+const DEFAULT_ERROR_MESSAGE = 'Injecting this service is not allowed from this file.';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow injecting certain services under certain paths',
+      category: 'Miscellaneous',
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-restricted-service-injections.md',
+    },
+    schema: {
+      type: 'array',
+      minItems: 1,
+      items: [
+        {
+          type: 'object',
+          required: ['paths', 'services'],
+          properties: {
+            paths: {
+              type: 'array',
+              minItems: 1,
+              items: {
+                type: 'string',
+              },
+            },
+            services: {
+              type: 'array',
+              minItems: 1,
+              items: {
+                type: 'string',
+              },
+            },
+            error: {
+              type: 'string',
+            },
+          },
+          additionalProperties: false,
+        },
+      ],
+    },
+  },
+
+  DEFAULT_ERROR_MESSAGE,
+
+  create(context) {
+    // Validate options.
+    context.options.forEach((option) =>
+      option.services.forEach((service) =>
+        assert(
+          service.toLowerCase() === service,
+          'Service name should be passed in kebab-case (all lower case)'
+        )
+      )
+    );
+
+    // Find matching blacklist entries for this file path.
+    const blacklists = context.options.filter((option) =>
+      option.paths.some((path) => context.getFilename().match(path))
+    );
+
+    if (!blacklists || blacklists.length === 0) {
+      return {};
+    }
+
+    function checkForViolationAndReport(node, serviceName) {
+      const serviceNameKebabCase = serviceName.split('/').map(kebabCase).join('/'); // splitting is used to avoid converting folder/ to folder-
+
+      blacklists.forEach((blacklist) => {
+        // Blacklist services are always passed in in kebab-case, so we can do a kebab-case comparison.
+        if (blacklist.services.includes(serviceNameKebabCase)) {
+          context.report({
+            node,
+            message: blacklist.error || DEFAULT_ERROR_MESSAGE,
+          });
+        }
+      });
+    }
+
+    return {
+      // Handles:
+      // * myService: service()
+      // * propertyName: service('myService')
+      Property(node) {
+        if (!emberUtils.isInjectedServiceProp(node)) {
+          return;
+        }
+
+        const callExpression = node.value;
+
+        // Get the service name either from the string argument or from the property name.
+        if (callExpression.arguments && callExpression.arguments.length === 1) {
```

Comment:
If they have more than one argument, is it completely invalid? Or would it be safer to check `>= 1` here?

Reply:
This function only accept one argument, but I'll change to `>= 1` just in case.

https://api.emberjs.com/ember/release/functions/@ember%2Fservice/inject